spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -255,10 +255,20 @@ object DistriOptimizer extends AbstractOptimizer {
               val localCriterion = cached.localCriterions(i)
               val input = miniBatchBuffer(i).getInput()
               val target = miniBatchBuffer(i).getTarget()
-              val output = localModel.forward(input)
-              lossArray(i) = ev.toType[Double](localCriterion.forward(output, target))
-              val errors = localCriterion.backward(output, target)
-              localModel.backward(input, errors)
+              Engine.getEngineType() match {
+                case MklBlas =>
+                  val output = localModel.forward(input)
+                  lossArray(i) = ev.toType[Double](localCriterion.forward(output, target))
+                  val errors = localCriterion.backward(output, target)
+                  localModel.backward(input, errors)
+                case MklDnn =>
+                  Engine.computing.invokeAndWait2(Array(0).map(_ => () => {
+                    val output = localModel.forward(input)
```

Comment:
if localModel is IRgraph, there will be computing.invokeAndWait2 calling computing.invokeAndWait2 in IRGraph. Is it ok?

Reply:
No. Here we only consider the raw dnn model.

Reply:
I will add some condition to handle the `IRGraph`.